docs/Rest-API.md

```diff
@@ -0,0 +1,147 @@
+# Axe REST API
+
+Documentation of the REST API that comes along with the Axe service.
+
+## AxeResult
+
+This endpoint serves as a wrapper to collect all of the information from an axe run. 
+
+### Motivation
+
+- Provide access to results and context easily.
+- Avoid timing issues that come with multiple requests.
+
+### Simplified Sample
+```
+{
+  "axeContext" : { @AxeContext },
+  "axeResult" : { AXRL },
+  "axeConf" : { TBD by Framework }  
+}
+```
+
+### Requirements
+
+- Data points are collected at the same point in time, or as close as can be expected.
+
+## AxeContext
+
+Wraps data together that is needed to identify a particular View, with a particular screenshot,
+for a particular device configuration.
+
+### Motivation
+
+- Collect View information in one place.
+- Allow for direct correlation of screenshot and ViewHierarchy
+- Easy developer debugging.
+- Reproducability of results.
+
+### Simplified Sample
+
+```
+{
+  "axeView" : @AxeView,
+  "axeDevice" : @AxeDevice,
+  "axeEvents" : @AxeEvents,
+  "screenshot" : "base64png data"
+}
+```
+
+## AxeView
+
+Serves up properties about the Accessibility Hierarchy of a particular view. 
+
+### Motivation
+
+- All motivations of AxeContext apply.
+- Rule debugging.
+- Simplified test case collection.
+- Simplified false positive identification.
+
+### Simplified Sample
+
+```
+{
```

Comment:
How will elements be identified in AXRL? It seems like a path-based approach would be possible with this view representation; what are thoughts about an ID for each node in the tree?

Reply:
The DRY principle does not apply to documentation. Karan's question is valuable, and indicates information which is not readily apparent and which, at the very least, should be referenced.

Reply:
You're lacking context in which this document will actually be published. I understand your concern, if it's still a concern when this gets published we will address it then, for now, we should drop this conversation.